build/mongodl.py

```diff
@@ -0,0 +1,666 @@
+"""
+Download and extract MongoDB components.
+
+Use '--help' for more information.
+"""
+import argparse
+import enum
+import hashlib
+import json
+import os
+import platform
+import re
+import shutil
+import sqlite3
+import sys
+import tarfile
+import tempfile
+import textwrap
+import urllib.error
+import urllib.request
+import zipfile
+from collections import namedtuple
+from contextlib import contextmanager
+from fnmatch import fnmatch
+from pathlib import Path, PurePath, PurePosixPath
+
+DISTRO_ID_MAP = {
+    'elementary': 'ubuntu',
+    'fedora': 'rhel',
+    'centos': 'rhel',
+    'mint': 'ubuntu',
+    'opensuse-leap': 'sles',
+    'opensuse': 'sles',
+    'redhat': 'rhel',
+}
+
+DISTRO_VERSION_MAP = {
+    'elementary': {
+        '6': '20.04'
+    },
+}
+
+DISTRO_ID_TO_TARGET = {
+    'ubuntu': {
+        '20.*': 'ubuntu2004',
+    },
+    'debian': {
+        '9': 'debian92',
+        '10': 'debian10',
+        '11': 'debian11',
+    },
+    'rhel': {
+        '6': 'rhel62',
+        '7': 'rhel73',
+        '8': 'rhel81',
+    },
+    'sles': {
+        '10.*': 'suse10',
+        '11.*': 'suse11',
+        '12.*': 'suse12',
+        '13.*': 'suse13',
+        '15.*': 'suse15',
+    },
+    'amzn': {
+        '2018': 'amzn64',
+        '2': 'amzn64',
+    },
+}
+
+
+def infer_target():
+    if sys.platform == 'win32':
+        return 'windows'
+    if sys.platform == 'darwin':
+        return 'macos'
+    # Now the tricky bit
+    if Path('/etc/os-release').is_file():
+        return _infer_target_os_rel()
+    raise RuntimeError("Don't know yet how to find the default download "
+                       "'--target' for this system. Please contribute!")
```

Comment:
```suggestion
    raise RuntimeError(
        "We don't know how to find the default download '--target' on this system. Please contribute!")
```
Minor wording improvement suggestion for consistency with other error messages.